frontends/smvparser.l

```diff
@@ -0,0 +1,131 @@
+%{
+  #include"smv_node.h"
+  #include "smv_encoder.h"
+  #include "smvparser.h" 
+  using namespace std;
+  int line_num = 1;
+  #include <string>
+  #undef yyterminate
+  #define yyterminate()  return(cosa::smvparser::make_END())
+%}
+
+%option yylineno
+%option noyywrap
+%x COMMENT
+%x WORD_INIT WORD_BASE WORD_VALUE
+%%
+
+[ \t]     ;
+MODULE {return cosa::smvparser::make_MODULE();}
+main  {return cosa::smvparser::make_tok_main();}
+IVAR   {return cosa::smvparser::make_IVAR();}
+VAR  {return cosa::smvparser::make_VAR();}
+INVAR   {return cosa::smvparser::make_INVAR();}
+FROZENVAR   {return cosa::smvparser::make_FROZENVAR();}
+INVARSPEC   {return cosa::smvparser::make_INVARSPEC();}
+INIT   {return cosa::smvparser::make_INIT();}
+TRANS   {return cosa::smvparser::make_TRANS();}
+READ   {return cosa::smvparser::make_READ();}
+WRITE   {return cosa::smvparser::make_WRITE();}
+ASSIGN   {return cosa::smvparser::make_ASSIGN();}
+CONSTARRAY   {return cosa::smvparser::make_CONSTARRAY();}
+CONSTANTS   {return cosa::smvparser::make_CONSTANTS();}
+FUN   {return cosa::smvparser::make_FUN();}
+DEFINE {return cosa::smvparser::make_DEFINE();}
+
+next  {return cosa::smvparser::make_tok_next();}
+init {return cosa::smvparser::make_tok_init();}
+
+TRUE   {return cosa::smvparser::make_TOK_TRUE(true);}
+FALSE   {return cosa::smvparser::make_TOK_FALSE(false);}
+
+"signed word" {return cosa::smvparser::make_signed_word();}
+"unsigned word" {return cosa::smvparser::make_unsigned_word();}
+"array word"   {return cosa::smvparser::make_arrayword();}
+"array integer"   {return cosa::smvparser::make_arrayinteger();}
+"array" {return cosa::smvparser::make_tok_array();}
+
+pi {return cosa::smvparser::make_pi();}
+abs {return cosa::smvparser::make_ABS();}
+max {return cosa::smvparser::make_MAX();}
+min {return cosa::smvparser::make_MIN();}
+sin {return cosa::smvparser::make_SIN();}
+cos {return cosa::smvparser::make_COS();}
+exp {return cosa::smvparser::make_EXP();}
+tan {return cosa::smvparser::make_TAN();}
+ln {return cosa::smvparser::make_ln();}
+of  {return cosa::smvparser::make_of();}
+word1 {return cosa::smvparser::make_word1();}
+bool {return cosa::smvparser::make_tok_bool();}
+toint   {return cosa::smvparser::make_tok_toint();}
+count {return cosa::smvparser::make_tok_count();}
+swconst {return cosa::smvparser::make_swconst();}
+uwconst {return cosa::smvparser::make_uwconst();}
+sizeof {return cosa::smvparser::make_tok_sizeof();}
+floor {return cosa::smvparser::make_tok_floor();}
+extend {return cosa::smvparser::make_extend();}
+resize {return cosa::smvparser::make_resize();}
+typeof {return cosa::smvparser::make_tok_typeof();}
+
+unsigned   {return cosa::smvparser::make_tok_unsigned();}
+signed   {return cosa::smvparser::make_tok_signed();}
+boolean   {return cosa::smvparser::make_bool_type();}
+real  {return cosa::smvparser::make_real_type();}
+integer   {return cosa::smvparser::make_integer_type();}
+ck   {return cosa::smvparser::make_clock_type();}
```

Comment:
do we want to support clocks?
clocks are used to model timed-transition systems